ffmpeg/api_test.go

```diff
@@ -1288,6 +1288,42 @@ func TestTranscoder_OutputFPS(t *testing.T) {
 	outputFPS(t, Software)
 }
 
+func TestTranscoderAPI_ClipInvalidConfig(t *testing.T) {
+	// Test the following file open results on input: fail, success, fail, success
+
+	tc := NewTranscoder()
+	defer tc.StopTranscoder()
+	in := &TranscodeOptionsIn{}
+	out := []TranscodeOptions{{
+		Oname:        "-",
+		VideoEncoder: ComponentOptions{Name: "drop"},
+		From:         time.Second,
+	}}
+
+	_, err := tc.Transcode(in, out)
+	if err == nil || err != ErrTranscoderClipConfig {
+		t.Errorf("Expected '%s', got %v", ErrTranscoderClipConfig, err)
+	}
+	out[0].VideoEncoder.Name = "copy"
+	_, err = tc.Transcode(in, out)
+	if err == nil || err != ErrTranscoderClipConfig {
+		t.Errorf("Expected '%s', got %v", ErrTranscoderClipConfig, err)
+	}
+	out[0].From = 0
+	out[0].To = time.Second
+	_, err = tc.Transcode(in, out)
+	if err == nil || err != ErrTranscoderClipConfig {
+		t.Errorf("Expected '%s', got %v", ErrTranscoderClipConfig, err)
```

Comment:
Is clipping interval 0.0~1.0 an invalid parameter?

Reply:
That test case has set `copy` as encoder - we don't allow clipping for this case.